[2020-02-18 12:22:25,456] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 575 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-02-18 13:49:54,444] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-02-18 13:49:54,880] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-02-18 13:49:55,396] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-02-18 13:49:55,396] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-02-18 13:49:55,396] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-02-18 13:49:55,396] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-02-18 13:49:55,957] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 550 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-02-18 13:49:56,162] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-02-18 13:49:57,782] INFO Opening socket connection to server localhost/127.0.0.1:2181. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:50:16,219] INFO Socket connection established to localhost/127.0.0.1:2181, initiating session (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:50:43,945] WARN Client session timed out, have not heard from server in 28597ms for sessionid 0x100006d14f90040 (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:50:43,947] INFO Client session timed out, have not heard from server in 28597ms for sessionid 0x100006d14f90040, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:50:46,684] INFO Opening socket connection to server localhost/127.0.0.1:2181. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:50:46,684] INFO Socket connection established to localhost/127.0.0.1:2181, initiating session (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:50:47,006] WARN Unable to reconnect to ZooKeeper service, session 0x100006d14f90040 has expired (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:50:47,016] INFO Unable to reconnect to ZooKeeper service, session 0x100006d14f90040 has expired, closing socket connection (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:50:48,439] INFO [ZooKeeperClient] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2020-02-18 13:50:49,549] INFO EventThread shut down for session: 0x100006d14f90040 (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:50:49,870] INFO [ZooKeeperClient] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2020-02-18 13:50:49,954] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=6000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@1623b78d (org.apache.zookeeper.ZooKeeper)
[2020-02-18 13:50:51,027] INFO Opening socket connection to server localhost/127.0.0.1:2181. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:50:51,076] INFO Socket connection established to localhost/127.0.0.1:2181, initiating session (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:50:52,594] INFO Session establishment complete on server localhost/127.0.0.1:2181, sessionid = 0x100006d14f90041, negotiated timeout = 6000 (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:51:04,098] WARN Client session timed out, have not heard from server in 9361ms for sessionid 0x100006d14f90041 (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:51:04,347] INFO Client session timed out, have not heard from server in 9361ms for sessionid 0x100006d14f90041, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:51:05,887] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2020-02-18 13:51:06,502] INFO Opening socket connection to server localhost/0:0:0:0:0:0:0:1:2181. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:51:06,504] INFO Socket connection established to localhost/0:0:0:0:0:0:0:1:2181, initiating session (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:51:06,630] INFO EventThread shut down for session: 0x100006d14f90041 (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:51:06,672] INFO [ZooKeeperClient] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2020-02-18 13:51:06,672] WARN Unable to reconnect to ZooKeeper service, session 0x100006d14f90041 has expired (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:51:06,673] INFO Unable to reconnect to ZooKeeper service, session 0x100006d14f90041 has expired, closing socket connection (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:51:07,089] ERROR Error while creating ephemeral at /brokers/ids/0 with return code: SESSIONEXPIRED (kafka.zk.KafkaZkClient$CheckedEphemeral)
[2020-02-18 13:51:07,090] INFO Result of znode creation at /brokers/ids/0 is: SESSIONEXPIRED (kafka.zk.KafkaZkClient)
[2020-02-18 13:51:10,672] INFO [ZooKeeperClient] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2020-02-18 13:51:10,706] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=6000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@1623b78d (org.apache.zookeeper.ZooKeeper)
[2020-02-18 13:51:10,910] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2020-02-18 13:51:10,998] INFO Opening socket connection to server localhost/127.0.0.1:2181. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:51:10,998] INFO Socket connection established to localhost/127.0.0.1:2181, initiating session (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:51:11,138] INFO Session establishment complete on server localhost/127.0.0.1:2181, sessionid = 0x100006d14f90042, negotiated timeout = 6000 (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:51:11,717] INFO Result of znode creation at /brokers/ids/0 is: OK (kafka.zk.KafkaZkClient)
[2020-02-18 13:51:11,769] INFO Registered broker 0 at path /brokers/ids/0 with addresses: ArrayBuffer(EndPoint(dl12,9092,ListenerName(PLAINTEXT),PLAINTEXT)) (kafka.zk.KafkaZkClient)
[2020-02-18 13:52:01,460] WARN Client session timed out, have not heard from server in 15031ms for sessionid 0x100006d14f90042 (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:52:01,938] INFO Client session timed out, have not heard from server in 15031ms for sessionid 0x100006d14f90042, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:52:03,831] INFO Opening socket connection to server localhost/127.0.0.1:2181. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:52:03,831] INFO Socket connection established to localhost/127.0.0.1:2181, initiating session (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:52:04,174] WARN Unable to reconnect to ZooKeeper service, session 0x100006d14f90042 has expired (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:52:04,174] INFO Unable to reconnect to ZooKeeper service, session 0x100006d14f90042 has expired, closing socket connection (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:52:04,174] INFO EventThread shut down for session: 0x100006d14f90042 (org.apache.zookeeper.ClientCnxn)
[2020-02-18 13:52:04,495] INFO [ZooKeeperClient] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2020-02-18 13:52:07,240] ERROR Caught unexpected throwable (org.apache.zookeeper.ClientCnxn)
java.lang.NullPointerException
	at kafka.zookeeper.ZooKeeperClient$$anon$10.processResult(ZooKeeperClient.scala:234)
	at org.apache.zookeeper.ClientCnxn$EventThread.processEvent(ClientCnxn.java:633)
	at org.apache.zookeeper.ClientCnxn$EventThread.queuePacket(ClientCnxn.java:488)
	at org.apache.zookeeper.ClientCnxn.finishPacket(ClientCnxn.java:658)
	at org.apache.zookeeper.ClientCnxn.conLossPacket(ClientCnxn.java:676)
	at org.apache.zookeeper.ClientCnxn.queuePacket(ClientCnxn.java:1454)
	at org.apache.zookeeper.ZooKeeper.multiInternal(ZooKeeper.java:991)
	at org.apache.zookeeper.ZooKeeper.multi(ZooKeeper.java:935)
	at kafka.zookeeper.ZooKeeperClient.send(ZooKeeperClient.scala:232)
	at kafka.zookeeper.ZooKeeperClient$$anonfun$handleRequests$1$$anonfun$apply$1.apply$mcV$sp(ZooKeeperClient.scala:145)
	at kafka.zookeeper.ZooKeeperClient$$anonfun$handleRequests$1$$anonfun$apply$1.apply(ZooKeeperClient.scala:145)
	at kafka.zookeeper.ZooKeeperClient$$anonfun$handleRequests$1$$anonfun$apply$1.apply(ZooKeeperClient.scala:145)
	at kafka.utils.CoreUtils$.inLock(CoreUtils.scala:251)
	at kafka.utils.CoreUtils$.inReadLock(CoreUtils.scala:257)
	at kafka.zookeeper.ZooKeeperClient$$anonfun$handleRequests$1.apply(ZooKeeperClient.scala:144)
	at kafka.zookeeper.ZooKeeperClient$$anonfun$handleRequests$1.apply(ZooKeeperClient.scala:141)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at kafka.zookeeper.ZooKeeperClient.handleRequests(ZooKeeperClient.scala:141)
	at kafka.zk.KafkaZkClient.retryRequestsUntilConnected(KafkaZkClient.scala:1596)
	at kafka.zk.KafkaZkClient.kafka$zk$KafkaZkClient$$retryRequestUntilConnected(KafkaZkClient.scala:1589)
	at kafka.zk.KafkaZkClient.deletePartitionReassignment(KafkaZkClient.scala:831)
	at kafka.controller.KafkaController.removePartitionsFromReassignedPartitions(KafkaController.scala:845)
	at kafka.controller.KafkaController.kafka$controller$KafkaController$$maybeTriggerPartitionReassignment(KafkaController.scala:618)
	at kafka.controller.KafkaController.onControllerFailover(KafkaController.scala:263)
	at kafka.controller.KafkaController.kafka$controller$KafkaController$$elect(KafkaController.scala:1221)
	at kafka.controller.KafkaController$Reelect$.process(KafkaController.scala:1508)
	at kafka.controller.KafkaController$RegisterBrokerAndReelect$.process(KafkaController.scala:1517)
	at kafka.controller.ControllerEventManager$ControllerEventThread$$anonfun$doWork$1.apply$mcV$sp(ControllerEventManager.scala:89)
	at kafka.controller.ControllerEventManager$ControllerEventThread$$anonfun$doWork$1.apply(ControllerEventManager.scala:89)
	at kafka.controller.ControllerEventManager$ControllerEventThread$$anonfun$doWork$1.apply(ControllerEventManager.scala:89)
	at kafka.metrics.KafkaTimer.time(KafkaTimer.scala:31)
	at kafka.controller.ControllerEventManager$ControllerEventThread.doWork(ControllerEventManager.scala:88)
	at kafka.utils.ShutdownableThread.run(ShutdownableThread.scala:82)
[2020-02-18 13:52:23,801] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-02-18 13:56:10,233] WARN Attempting to send response via channel for which there is no open connection, connection id 172.16.112.12:9092-172.16.112.12:42192-79 (kafka.network.Processor)
[2020-02-18 13:56:10,295] WARN Attempting to send response via channel for which there is no open connection, connection id 172.16.112.12:9092-172.16.112.12:42198-80 (kafka.network.Processor)
[2020-02-18 13:56:28,358] WARN Attempting to send response via channel for which there is no open connection, connection id 172.16.112.12:9092-172.16.112.12:42200-80 (kafka.network.Processor)
[2020-02-18 13:56:28,361] WARN Attempting to send response via channel for which there is no open connection, connection id 172.16.112.12:9092-172.16.112.12:42194-79 (kafka.network.Processor)
[2020-02-18 13:56:29,621] WARN Attempting to send response via channel for which there is no open connection, connection id 172.16.112.12:9092-172.16.112.12:42196-79 (kafka.network.Processor)
[2020-02-18 13:56:29,621] WARN Attempting to send response via channel for which there is no open connection, connection id 172.16.112.12:9092-172.16.112.12:42188-78 (kafka.network.Processor)
